pilot/pkg/model/authentication.go

```diff
@@ -238,6 +240,9 @@ func getConfigsForWorkload(rootNamespace string, configsByNamespace map[string][
 	} else {
 		lookupInNamespaces = []string{namespace}
 	}
+	if serviceNamespace != "" && !slices.Contains(lookupInNamespaces, serviceNamespace) {
+		lookupInNamespaces = append(lookupInNamespaces, serviceNamespace)
```

Comment:
RequestAuthentication is in same ns with workload. Not to mention the selector field, so i believe this is not expected to work well

Reply:
This should be a no-op if the RA is in the same ns as the workload/waypoint but in the cross-namespace waypoint case the RA would be in the same ns as the service/se it's targeting and the waypoint would be in a different namespace. This is the setup which is meant to be covered by this code.

Reply:
Yeah, i know this case. But if a ra is set with selector, how do we match?

Reply:
We shouldn't match. Selector shouldn't bind to waypoint.

Reply:
OK, so it means we cannot specify selector in ambient, only targetRef is supported, right?

Reply:
yup, waypoints only support gtwapi-style policy attachment.